toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -761,21 +762,28 @@ func updateInitramfsForEncrypt(installChroot *safechroot.Chroot) (err error) {
 	return
 }
 
-func updateFstab(installRoot string, installMap, mountPointToFsTypeMap, mountPointToMountArgsMap map[string]string) (err error) {
+func updateFstab(installRoot string, installMap, mountPointToFsTypeMap, mountPointToMountArgsMap map[string]string, hidepidEnable bool) (err error) {
 	ReportAction("Configuring fstab")
 
 	for mountPoint, devicePath := range installMap {
 		if mountPoint != "" && devicePath != NullDevice {
-			err = addEntryToFstab(installRoot, mountPoint, devicePath, mountPointToFsTypeMap[mountPoint], mountPointToMountArgsMap[mountPoint])
+			err = addEntryToFstab(installRoot, mountPoint, devicePath, mountPointToFsTypeMap[mountPoint], mountPointToMountArgsMap[mountPoint], false)
 			if err != nil {
 				return
 			}
 		}
 	}
+
+	if hidepidEnable {
+		err = addEntryToFstab(installRoot, "/proc", "proc", "proc", "rw,nosuid,nodev,noexec,relatime,hidepid=2", true)
```

Comment:
Won't this be a problem if there already was an entry for `/proc` inside `installMap`?

Reply:
/proc should never be added to the installMap. InstallMap gets its data the following way
```partIDToDevPathMap --> mountPointDevPathMap  == mountPointMap --> installMap``` 
Therefore, since `/proc ` never has a partition, it is not added. That being said, a user could add /proc to the fstab using postinstall scripts which would be a problem. I should document that case.

Reply:
Please do, thank you!